### PR TITLE
Add support for skip pattern

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves: #16 

Example implementation: https://github.com/smallrye/smallrye-opentracing/pull/23. The implementation itself is in https://github.com/opentracing-contrib/java-jaxrs/blob/master/opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java#L128

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

File: tck/src/main/java/org/eclipse/microprofile/opentracing/tck/OpenTracingBaseTests.java [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.resolver.api.maven.Maven;
+import org.testng.Assert;
+import org.testng.Reporter;
+import org.testng.annotations.BeforeMethod;
+
+/**
+ * @author Pavol Loffay
+ */
+public class OpenTracingBaseTests extends Arquillian {
```
Comment (pavolloffay) [inline tck/src/main/java/org/eclipse/microprofile/opentracing/tck/OpenTracingBaseTests.java]: The `OpentracingClientTests` is quite big I have introduced this base class with some common methods. Different features can be tested in separate test classes.

**Comment by pavolloffay** [comment]
@pilhuhn @fmhwong @objectiser could you please review?

**Comment by pavolloffay** [comment]
thanks @objectiser, I have changed year in license headers in new files to 2018. There is a license check and it was passing even with 2017 so I guess it's fine.

File: tck/src/main/java/org/eclipse/microprofile/opentracing/tck/OpenTracingClientTests.java [hunk 2]
```diff
@@ -74,50 +55,7 @@
  * Opentracing TCK tests.
  * @author <a href="mailto:steve.m.fontes@gmail.com">Steve Fontes</a>
  */
-public class OpentracingClientTests extends Arquillian {
```
Comment (pavolloffay) [inline tck/src/main/java/org/eclipse/microprofile/opentracing/tck/OpenTracingClientTests.java]: I have renamed it to `OpenTracing`

**Comment by fmhwong** [comment]
LGTM

**Pull request merged**